server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -199,6 +207,31 @@ public DiskThresholdSettings(Settings settings, ClusterSettings clusterSettings)
         clusterSettings.addSettingsUpdateConsumer(CLUSTER_ROUTING_ALLOCATION_DISK_THRESHOLD_ENABLED_SETTING, this::setEnabled);
     }
 
+    public static DiskThresholdSettings readFrom(StreamInput in) throws IOException {
+        DiskThresholdSettings diskThresholdSettings = new DiskThresholdSettings();
+        diskThresholdSettings.setLowWatermark(RelativeByteSizeValue.readFrom(in));
```

Comment:
I have a slight preference for not adding a new empty constructor here and instead constructing a suitable `Settings` (and empty `ClusterSettings`) to pass into the existing constructor.

Reply:
I would also be ok with adding a new constructor that accepts all the values at once, I just don't think we should allow constructing a `DiskThresholdSettings` with null values for its fields.